### ISSUE TITLE
False Positive: invalid-name for conditional constants with 4.0.2

### ISSUE THREAD

**Issue by avylove** [pylint-dev/pylint#10697]
### Bug description

It seems like 4.0.2 fixed some cases, but this combined case still fails. Minimized example taken from [Enlighten](https://github.com/Rockhopper-Technologies/enlighten).

```python
"""Simple Pylint Test"""

import platform
import sys

if platform.system() == 'Windows':
    SERIES_STD = u' ▌█'
else:
    SERIES_STD = u' ▏▎▍▌▋▊▉█'

try:
    SERIES_STD.encode(sys.__stdout__.encoding)
except UnicodeEncodeError:
    SERIES_STD = u' |'
except (AttributeError, TypeError):
    pass
```

### Command used

```shell
pylint pylint_4_0_2_issue.py
```

### Pylint output

```python
pylint_4_0_2_issue.py:7:4: C0103: Variable name "SERIES_STD" doesn't conform to snake_case naming style (invalid-name)
pylint_4_0_2_issue.py:9:4: C0103: Variable name "SERIES_STD" doesn't conform to snake_case naming style (invalid-name)
pylint_4_0_2_issue.py:14:4: C0103: Variable name "SERIES_STD" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

No errors

### Pylint version

```shell
pylint 4.0.2
astroid 4.0.1
Python 3.13.7 (main, Aug 14 2025, 00:00:00) [GCC 15.2.1 20250808 (Red Hat 15.2.1-1)]
```

### OS / Environment

Fedora 42

### Additional dependencies

```python

```

**Comment by jacobtylerwalls** [comment]
Hi @avylove. This is expected, since the variable is assigned twice, once during the if/else, and a second time during the except. I know this is probably not what you want, so you have a few options:
- disabling
- calculating the value first and then assigning to an upper-case constant only once
